api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -92,4 +96,65 @@ public Person createFHIRPerson(Person person) {
 		return FHIRPersonUtil.generatePerson(omrsPerson);
 	}
 	
+	@Override
+	public Person updateFHIRPerson(Person thePerson, String theId) {
+		org.openmrs.api.PersonService personService = Context.getPersonService();
+		org.openmrs.Person retrievedPerson = personService.getPersonByUuid(theId);
+		if (retrievedPerson != null) { // update person
+			org.openmrs.Person omrsPerson = FHIRPersonUtil.generateOpenMRSPerson(thePerson);
+			retrievedPerson = updatePersonAttributes(omrsPerson, retrievedPerson);
+			Context.getPersonService().savePerson(retrievedPerson);
+			return FHIRPersonUtil.generatePerson(retrievedPerson);
+		} else { // no person is associated with the given uuid. so create a new person with the given uuid
+			if (thePerson.getId() == null) { // since we need to PUT the Person to a specific URI, we need to set the uuid here, if it is not already set.
+				IdDt uuid = new IdDt();
+				uuid.setValue(theId);
+				thePerson.setId(uuid);
+			}
+			return createFHIRPerson(thePerson);
+		}
+	}
+	
+	public org.openmrs.Person updatePersonAttributes(org.openmrs.Person omrsPerson, org.openmrs.Person retrievedPerson) {
```

Comment:
scope of the method should be private. Will you think it will be useful? If that so please mve this to util class